### DISCUSSION TITLE
Deep filter relationships

### DISCUSSION THREAD

**Discussion by perostlin** [pocketbase/pocketbase#2720] [Q&A] [answered]
**Hello**,

I have the following structure in my Pocketbase:
- **Tournaments**
- **Teams**
- **Users**

And the following relationship structure:
- **Tournaments** - _many-to-many_ - **Teams**
- **Teams** - _many-to-many_ - **Users**

**Question**:
I'm trying to filter out Tournaments where a specific User is a member of a Team.

```js
const tournaments = await locals.pocketbase.collection('tournaments').getFullList({
    filter: `teams.members ~ '${user.id}'`,
    expand: 'teams,teams.members'
});
```

<img width="1425" alt="image" src="https://github.com/pocketbase/pocketbase/assets/20464832/1c04fd95-c863-4e7d-845e-45f4e5eae811">

**Comment by perostlin** [comment]
Tried to create a '**Single**' type relationship instead and do the above filtration and that worked.

Doesn't work:
<img width="650" alt="image" src="https://github.com/pocketbase/pocketbase/assets/20464832/603bd82c-f146-44ad-a08d-b0f7e467cad9">

Works:
<img width="655" alt="image" src="https://github.com/pocketbase/pocketbase/assets/20464832/a0545315-cd8c-4ec3-b44f-ce8174c74323">

**Comment by ganigeorgiev** [comment, marked as answer]
When the relation field is "multiple" we perform a match-all by default, aka. `teams.members ~ '${user.id}'` will require the `members` in every `teams` record related to a single tournament to have the specified user id.

If I'm understanding correctly your use case, you want the specified user to exist as member in at least one of the `tournaments`s `teams`, right?

In that case, you can prefix your operator with `?`, aka:

```js
teams.members ?~ '${user.id}'

// or (this could be slightly more performant as it is direct match)

teams.members.id ?= '${user.id}'
```

**Reply by perostlin** [reply]
Thank you for pointing me in the right direction!

The use case is to retrieve all tournaments where the user is part of a team participating in the retrieved tournaments.


The solution was to use '?' together with the '~':
?~ Any/At least one of Like/Contains (if not specified auto wraps the right string OPERAND in a "%" for wildcard match)
